recipes/slave.rb

```diff
@@ -110,9 +110,9 @@ class Chef::Recipe
     zookeeper_server_list: zk_server_list,
     zookeeper_port: zk_port,
     zookeeper_path: zk_path,
-    logs_dir: node['mesos']['logs_dir'],
-    work_dir: node['mesos']['work_dir'],
-    isolation_type: node['mesos']['isolation_type']
+    logs_dir: node['mesos']['slave']['env']['LOGS'],
```

Comment:
If it reads ```logs_dir: node['mesos']['slave']['flags']['log_dir']``` and you have the flag defined correctly, the MESOS_log_dir env variable will be set and it will be read directly by mesos.